scripts/dts/extract/compatible.py

```diff
@@ -43,6 +43,17 @@ def extract(self, node_address, yaml, prop, def_label):
         for i, comp in enumerate(compatible):
             edts_insert_device_property(device_id, 'compatible/{}'.format(i),
 					comp)
+            try:
+                device_types = yaml[comp]['type']
+            except KeyError:
+                raise Exception("Device Node: {} has no 'type'".format(node_address))
+
+            if not isinstance(device_types, list):
+                device_types = [device_types, ]
+            for j, dev_type in enumerate(device_types):
+                edts_insert_device_type(comp, dev_type)
```

Comment:
Remove - see above.

Reply:
After thinking a bit - device type generation should be moved  to heuristics.py.

Reply:
See reasonning behind this choice above

Reply:
My last answer was response to first comment.

> After thinking a bit - device type generation should be moved to heuristics.py.

Yes, I hesitated indeed. Moved